var/spack/repos/builtin/packages/dftd4/package.py

```diff
@@ -26,22 +29,38 @@ class Dftd4(MesonPackage):
     version("3.1.0", sha256="b652aa7cbf8d087c91bcf80f2d5801459ecf89c5d4176ebb39e963ee740ed54b")
     version("3.0.0", sha256="a7539d68d48d851bf37b79e37ea907c9da5eee908d0aa58a0a7dc15f04f8bc35")
 
-    depends_on("c", type="build")  # generated
-    depends_on("fortran", type="build")  # generated
+    depends_on("c", type="build")
+    depends_on("fortran", type="build")
 
     variant("openmp", default=True, description="Use OpenMP parallelisation")
-    variant("python", default=False, description="Build Python extension module")
+    variant(
+        "python",
+        default=False,
+        when="build_system=meson",
+        description="Build Python extension module",
+    )
+
+    depends_on("meson@0.57.1:", type="build", when="build_system=meson")  # mesonbuild/meson#8377
 
     depends_on("blas")
     depends_on("lapack")
-    depends_on("mctc-lib")
-    depends_on("meson@0.57.1:", type="build")  # mesonbuild/meson#8377
     depends_on("pkgconfig", type="build")
+
     depends_on("py-cffi", when="+python")
     depends_on("python@3.6:", when="+python")
 
+    depends_on("mctc-lib build_system=cmake", when="build_system=cmake")
```

Comment:
Why do you remove the dependency on mctc-lib for the meson build?

Reply:
I was under the impression that these dependencies are built as sub-projects by `meson`. I'm not very familiar with `meson`, so I might have misunderstood. Could you please clarify why `mctc-lib` is an explicit dependency, while `multicharge` is not (and wasn't even a Spack package)?

Reply:
I added explicit dependencies for both packages.